s4e-backend/src/main/java/pl/cyfronet/s4e/service/AppUserService.java

```diff
@@ -57,6 +79,26 @@ public AppUser createFromRequest(CreateUserWithGroupsRequest request, String ins
         return appUser;
     }
 
+    public EmailVerification confirmEmail(String token) throws NotFoundException, RegistrationTokenExpiredException, AppUserCreationException {
```

Comment:
You should declare `@Transactional`, the same as in ::save, as you call it from within the class (and so it won't have a transaction open). Moreover, it might make sense not to call ::save here, but rely on the hibernate for saving the entity as you are only updating it (the `@Transactional` is then essential here).